server/controllers/auth-controller.js

```diff
@@ -157,6 +161,106 @@ class UsersController {
       return errorResponse(res, error.message);
     }
   }
+
+  /**
+   * @static
+   * @param {*} req
+   * @param {*} res
+   * @memberof UsersController
+   * @returns {undefined}
+   */
+  static sendResetEmail(req, res) {
+    const { email } = req.body;
+
+    validate(req.body, passwordResetSchema)
+      .then(() => {
+        User.findOne({ where: { email } }).then(user => {
+          if (!user) {
+            return errorResponse(res, 'User not found!', 404);
+          }
+          const { username, id } = user;
+          const token = generateToken({ username, id });
+          mailer
+            .sendResetMail({
+              email,
+              username,
+              resetPasswordLink: generateResetLink(token)
+            })
+            .then(() => {
+              user.save();
```

Comment:
The user hasn't been modified at this point, no?

Reply:
it is totally redundant.

Reply:
So why is it still here?

Reply:
not there anymore :)